sdk/logical/request.go

```diff
@@ -108,6 +109,9 @@ type Request struct {
 	// authentication/protection.
 	Connection *Connection `json:"connection" structs:"connection" mapstructure:"connection"`
 
+	// ClientHeaderCert is the x509 client certificate passed from a header. nil if unset.
+	ClientHeaderCert *x509.Certificate
```

Comment:
```suggestion
	ClientHeaderCert []*x509.Certificate
```

might future-proof it and allow us to handle chains in the future.

Perhaps clarifying `certs[0]` is the leaf and `certs[1...]` are the intermediates, if any.

Reply:
(proposed alternative in https://github.com/openbao/openbao/pull/2080/files#r2511574637)